apps/remote_control/test/lexical/remote_control/code_intelligence/definition_test.exs

```diff
@@ -0,0 +1,370 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.DefinitionTest do
+  alias Lexical.RemoteControl.CodeIntelligence.Definition
+  alias Lexical.SourceFile
+  alias Lexical.SourceFile.Line
+  alias Lexical.SourceFile.Position
+
+  import Lexical.Test.Fixtures
+  import Lexical.Test.CodeSigil
+  import Line
+
+  use ExUnit.Case, async: false
+
+  setup do
+    start_supervised!(Lexical.SourceFile.Store)
+    Code.put_compiler_option(:ignore_module_conflict, true)
+    :ok
+  end
+
+  defp with_navigation_project(_ctx) do
+    %{project: project(:navigations)}
+  end
+
+  defp with_referenced_file(%{project: project}, file \\ "my_definition.ex") do
+    path = file_path(project, Path.join("lib", file))
+    Code.compile_file(path)
+    %{uri: SourceFile.Path.ensure_uri(path)}
+  end
+
+  defp open_uses_file(project, content) do
```

Comment:
the naming of this function is confusing, as is the return value.
calling it `source_file` seems a little better, and having it return `{:ok, source_file}` or `{:error, reason}` is better. 
Right now, it returns an error tuple or a naked source_file, making it hard to pattern match against.

Reply:
changed it to return `{:ok, uses_file} | {:error, reason}`

Reply:
i still think the `uses_file` is very awkward naming. I'd prefer `subject_module` something like that. Does that file need to actually exist? you're just replacing the file's contents with whatever is under test (which is fine). If you need a source file, just make one and pass that around.